lib/wrappers/rust/src/lib.rs

```diff
@@ -1,16 +1,35 @@
 use contourwall_core::{
     new, new_with_ports, show, single_new_with_port, solid_color, update_all, ContourWallCore,
 };
+
 use serialport::{self, available_ports};
 use std::{ffi::c_char, thread, time::Duration};
+
+/// # Description
+/// The ContourWall takse responsible for managing on how frame is pushed to the title via its members.
+///
+/// # Struct members
+///  - cw_core: having the type of ContourWallCore. The cw_core play the role of title administration with 2 functionalities.
+//   setting up the ContourWall with configurations and managing the sending buffer tasks following different type of commands.
+///
+///  - pixels: having the type of ndarray::Array3<u8> with the shape of 20x20x3. This is a 3D arrays that representS the pixels frame.
+///  
+///  - pushed_frame: an unsigned 32bit integer. this integer counts the number of frames that is pushed to the titles.
 pub struct ContourWall {
     cw_core: ContourWallCore,
     pub pixels: ndarray::Array3<u8>,
     pub pushed_frames: u32,
 }
 
 impl ContourWall {
-    pub fn new(baud_rate: u32) -> ContourWall {
+    ///Initialise the ContourWall, all the configuration and orchistration happens automatically.
+    ///
+    /// # Parameters
+    /// - Baudrate: an unsigned 32bit integer.
+    ///
+    /// # Returns
+    /// - The initialised ContourWall is return.
+    pub fn new_w(baud_rate: u32) -> ContourWall {
```

Comment:
You did this with every function, but I am only gonna comment it on this one.
Why did you call this `new_w` instead of `new`?

Reply:
initially, I guessed wrongly there is a bug occurring due to the same function name between the wrapper methods and the Contour Wall Core library, but it turned out that was not the case. Will fix this function name for better readability.